src/ucp/wireup/wireup_cm.c

```diff
@@ -731,11 +777,21 @@ static unsigned ucp_cm_server_connect_progress(void *arg)
  * Async callback on a server side which notifies that client is connected.
  */
 static void ucp_cm_server_connect_cb(uct_ep_h ep, void *arg,
-                                     ucs_status_t status)
+                                     uct_cm_ep_server_connect_cb_handle_t
+                                     *connect_cb_handle)
 {
     ucp_ep_h ucp_ep            = arg;
     uct_worker_cb_id_t prog_id = UCS_CALLBACKQ_ID_NULL;
     ucp_lane_index_t cm_lane;
+    ucs_status_t status;
+
+    if (!(connect_cb_handle->field_mask & UCT_CM_EP_SERVER_CONNECT_CB_HANDLE_STATUS)) {
+        ucs_error("Incorrect fields on server connect callback (0x%lx)",
+                  connect_cb_handle->field_mask);
+        status = UCS_ERR_IO_ERROR;
+    } else {
```

Comment:
better return immediately if there is error, not else

Reply:
without calling ucp_worker_set_ep_failed() ? why?

Reply:
maybe just replace all these error checks by assertions?